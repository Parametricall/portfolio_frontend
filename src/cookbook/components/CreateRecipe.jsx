import {
    Button, Container, makeStyles, TextField,
} from '@material-ui/core';
import React, { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import { Redirect } from 'react-router-dom';
import { fetchData } from '../../utilities';
import { CREATE_RECIPE_URL } from '../../constants';
import DynamicList from '../../components/DynamicList';
import Ingredient from './Ingredient';
import Method from './Method';
import RecipeName from './RecipeName';

export const emptyIngredient = () => ({
    id: null,
    food: {},
    quantity: '',
    __isNew__: true,
    __key__: uuidv4(),
});

export const emptyMethod = () => ({
    id: null,
    method: '',
    __isNew__: true,
    __key__: uuidv4(),
});

const useStyles = makeStyles(() => ({
    submitButton: {
        float: 'right',
    },
}));

function CreateRecipe() {
    const classes = useStyles();
    const [redirect, setRedirect] = useState(false);

    const [title, setTitle] = useState(null);
    const [ingredients, setIngredients] = useState([emptyIngredient()]);
    const [methods, setMethods] = useState([emptyMethod()]);

    const handleSubmit = async () => {
        const postData = {
            name: title,
            ingredients,
            methods,
        };
        await fetchData(CREATE_RECIPE_URL, 'POST', postData);
        setRedirect(true);
    };

    if (redirect) return <Redirect to="/cookbook" />;

    return (
        <Container maxWidth="md">
            <RecipeName value={title} onChange={setTitle} editable />
            <DynamicList
                valueList={ingredients}
                onChange={setIngredients}
                emptyChildCallback={emptyIngredient}
                idKey="__key__"
                Component={Ingredient}
            />
            <DynamicList
                valueList={methods}
                onChange={setMethods}
                emptyChildCallback={emptyMethod}
                idKey="__key__"
                Component={Method}
            />
            <Button
                className={classes.submitButton}
                variant="contained"
                color="primary"
                onClick={handleSubmit}
            >
                Submit
            </Button>
        </Container>
    );
}

export default CreateRecipe;
