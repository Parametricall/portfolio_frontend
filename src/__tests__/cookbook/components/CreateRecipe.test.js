import React from 'react';
import { shallow } from 'enzyme';
import CreateRecipe from '../../../cookbook/components/CreateRecipe';


describe('CreateRecipe', () => {
  const createRecipeWrapper = shallow(<CreateRecipe/>);

  it('renders correctly', () => {
    expect(createRecipeWrapper).toMatchSnapshot()
  })
});