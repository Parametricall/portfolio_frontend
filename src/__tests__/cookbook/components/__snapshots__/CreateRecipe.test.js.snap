// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateRecipe renders correctly 1`] = `
<Container
  className="mt-5"
  fluid={false}
>
  <Form
    inline={false}
  >
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Recipe Name
      </FormLabel>
      <FormControl
        onChange={[Function]}
        value=""
      />
    </FormGroup>
    <h3>
      Ingredients
    </h3>
    <hr />
    <h4>
      Food
    </h4>
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Food
      </FormLabel>
      <FormControl
        onChange={[Function]}
        value={Array []}
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Quantity
      </FormLabel>
      <FormControl
        onChange={[Function]}
        value={Array []}
      />
    </FormGroup>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Submit
    </Button>
  </Form>
</Container>
`;
